javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -121,6 +123,26 @@ object Integer {
       asInt(result)
   }
 
+  private lazy val clz32: js.Function1[Int, Int] = {
+    if (linkingInfo.esVersion >= ESVersion.ES2015 ||
+        js.typeOf(js.Dynamic.global.Math.clz32) == "function") {
+      js.Dynamic.global.Math.clz32.asInstanceOf[js.Function1[Int, Int]]
```

Comment:
As is, this produces significantly worse code, especially in ES2015+. We should make sure to keep producing good code for ES2015+.

Moving the `if (linkingInfo.esVersion >= ESVersion.ES2015)` part of the test to the `@inline def` body should achieve that. Something like
```scala
  @inline def numberOfLeadingZeros(i: scala.Int): scala.Int = {
    if (linkingInfo.esVersion >= ESVersion.ES2015)
      js.Dynamic.global.Math.clz32(i).asInstanceOf[Int]
    else
      clz32(i)
  }
```